interface/java_binding/src/test/java/org/khronos/ktx/test/KtxTexture2Test.java

```diff
@@ -227,4 +228,72 @@ public void testCreate() {
 
         texture.destroy();
     }
+
+	@Test
+	public void testInputSwizzleBasisEx() throws IOException {
+
+		// Create RGBA pixels for an image with 32x32 pixels,
+		// filled with
+		// 8 rows of red pixels
+		// 8 rows of green pixels
+		// 8 rows of blue pixels
+		// 8 rows of white pixels
+		int sizeX = 32;
+		int sizeY = 32;
+		byte[] rgba = new byte[sizeX * sizeY * 4];
+		fillRows(rgba, sizeX, sizeY, 0, 8, 255, 0, 0, 255); // Red
+		fillRows(rgba, sizeX, sizeY, 8, 16, 0, 255, 0, 255); // Green
+		fillRows(rgba, sizeX, sizeY, 16, 24, 0, 0, 255, 255); // Blue
+		fillRows(rgba, sizeX, sizeY, 24, 32, 255, 255, 255, 255); // White
+
+		// Create a texture and fill it with the RGBA pixel data
+		KtxTextureCreateInfo info = new KtxTextureCreateInfo();
+		info.setBaseWidth(sizeX);
+		info.setBaseHeight(sizeY);
+		info.setVkFormat(VkFormat.VK_FORMAT_R8G8B8A8_SRGB);
+		KtxTexture2 t = KtxTexture2.create(info, KtxCreateStorage.ALLOC);
+		t.setImageFromMemory(0, 0, 0, rgba);
+
+		// Apply basis compression with an input swizzle, BRGA, so that
+		// the former B channel becomes the R channel
+		// the former R channel becomes the G channel
+		// the former G channel becomes the B channel
+		// the former A channel remains the A channel
+		KtxBasisParams p = new KtxBasisParams();
+		p.setUastc(false);
+		p.setInputSwizzle(new char[] { 'b', 'r', 'g', 'a' });
+		t.compressBasisEx(p);
+
+		// Obtain the texture data, and compare it to the
+		// expected data of the reference texture
```

Comment:
Since you are creating the texture in the test, I'd prefer not to need a reference file. You can transcode the swizzled `t` to `KTX_TTF_RGBA32` and compare those pixels with the pixels in `rgba`. Given the full on colors you are using I don't think compression & transcode will change them.

Reply:
I'd also like to avoid that reference file. (One reason: The `tests/testimages` directory already contains >100 files, without any information about what these files are, where they come from, and what they are used for. I'd like to avoid throwing more images on that pile...)

And what you suggested might ... _nearly_ work: From a quick test, converting the resulting data to `KTX_TTF_RGBA32` gave the following input/output pxiel values:
```
Pixel 0 in input  255,   0,   0, 255
Pixel 0 in output   2, 255,   2, 255
...
Pixel 256 in input    0, 255,   0, 255
Pixel 256 in output   0,   0, 253, 255
...
Pixel 512 in input    0,   0, 255, 255
Pixel 512 in output 253,   0,   0, 255
```
The swizzling is applied correctly, but there is a _slight_ deviation in the formerly "pure" colors, of `+/- 2` in some components. I don't know where this might come from - maybe something about SRGB, or it's a compression artifact, or whatnot. If you have an idea what might be causing this (or how to avoid this), I could change it accordingly. 

Regarldless of that, the test _could_ be changed to not use a reference image, by comparing the result of converting the image to `KTX_TTF_RGBA32` to pixel values that are created _programmatically_ (using the same functions that are used for creating the input image).

This could then be 
- A **fixed** comparison to the values that I currently see in the output (`2, 255,   2`)
- Or a comparison with `expected=(0, 255, 0)` **but** with some "epsilon" of maybe `+/-5` to account for the slight changes...

Any preference?

Reply:
I prefer the fixed comparison as the test will not then be sensitive to changes in the encoder. You could also try `ktxTexture2_CompressAstcEx`. Perhaps that will have less artifacts.

Reply:
> You could also try ktxTexture2_CompressAstcEx. Perhaps that will have less artifacts.

Actually, in order to have proper test coverage, I can not only 'try' this (as an alternative), but actually have to test this _in addition_ to the current test. Both of them (basis and astc) are going through different code paths (even thought they are doing the same sort of conversion for the swizzling).